packages/gatsby-plugin-fastify/src/plugins/headers.ts

```diff
@@ -0,0 +1,46 @@
+import fp from "fastify-plugin";
+
+import { SECURITY_HEADERS } from "../utils/constants";
+import { removeQueryParmsFromUrl } from "../utils/routes";
+
+import type { Headers } from "../utils/headers";
+import type { FastifyPluginAsync, FastifyReply } from "fastify";
+
+export const handleHeaders: FastifyPluginAsync<{
+  headers: Headers;
+  useDefaultSecurity: boolean;
+}> = fp(async (fastify, { headers, useDefaultSecurity }) => {
```

Comment:
Also thought about logging...not a rush but before we merge we'll want a couple of debug logs noting the use of default security headers, default caching headers, and custom headers. custom headers can probably just say hey there are custom headers or aren't. see https://github.com/gatsby-uc/plugins/blob/main/packages/gatsby-plugin-fastify/src/plugins/client-routes.ts#L16 for an example.